src/interrupt.rs

```diff
@@ -76,24 +161,82 @@ pub unsafe fn enable() {
     }
 }
 
+/// Restore the global interrupt flag to its previous state before [crate::interrupt::disable_save].
+///
+/// *Hint*: Most of the time you probably don't want to use this function directly.
+///         Consider creating a critical section with [crate::interrupt::free] instead.
+///
+/// This function is an optimization fence.
+/// That means memory accesses will not be re-ordered by the compiler across this function call.
+///
+/// # Safety
+///
+/// - Do not call this function inside an [crate::interrupt::free] critical section
```

Comment:
Technically, you can call it inside a critical section if the corresponding `disable_save()` call was also inside the critical section.  I am really not sure how to express this, though :(

Reply:
I updated the comment and I also added a paragraph about nesting.